src/rtc.rs

```diff
@@ -0,0 +1,849 @@
+    /// as a number of microseconds rounded to the nearest whole number.
+    pub fn subsec_micros(&self) -> Option<u32> {
+        self.calendar_initialized()?;
+        let ss = self.reg.ssr.read().ss().bits();
+        Some(self.ss_to_us(ss))
+    }
+
+    /// Returns the raw value of the synchronous subsecond counter
+    ///
+    /// This counts up to `self.subsec_res()` then resets to zero once per second.
+    pub fn subsec_raw(&self) -> Option<u16> {
+        self.calendar_initialized()?;
+        Some(self.reg.ssr.read().ss().bits())
+    }
+
+    /// Returns the resolution of subsecond values
+    ///
+    /// The RTC counter increments this number of times per second.
+    pub fn subsec_res(&self) -> Option<u16> {
+        self.calendar_initialized()?;
+        Some(self.reg.prer.read().prediv_s().bits())
+    }
+
+    /// Returns the stored Daylight Savings Time status
+    pub fn dst(&self) -> DstState {
+        if self.reg.cr.read().bkp().bit_is_set() {
+            DstState::Dst
+        } else {
+            DstState::Standard
+        }
+    }
+
+    /// Sets the stored Daylight Savings Time status without adjusting the clock
+    pub fn set_dst(&mut self, dst: DstState) {
+        self.reg.cr.modify(|_, w| w.bkp().bit(dst == DstState::Dst));
+    }
+
+    /// Begin Daylight Savings Time
+    ///
+    /// Adds an hour to the stored time and sets the stored DST status.
+    /// Returns an error if the clock has not been initialized, or if the stored DST status
+    /// indicates it has already begun.
+    pub fn begin_dst(&mut self) -> Result<(), DstError> {
+        self.calendar_initialized()
+            .ok_or(DstError::ClockNotInitialized)?;
+        if self.reg.cr.read().bkp().bit_is_set() {
+            return Err(DstError::AlreadyDst);
+        }
+        self.reg
+            .cr
+            .modify(|_, w| w.add1h().set_bit().bkp().set_bit());
+        Ok(())
+    }
+
+    /// End Daylight Savings Time
+    ///
+    /// Subtracts an hour from the stored time and sets the stored DST status.
+    /// Returns an error if the clock has not been initialized, if the stored DST status
+    /// indicates it is already standard time, or if we cannot subtract an hour because
+    /// it is not at least one hour past midnight.
+    pub fn end_dst(&mut self) -> Result<(), DstError> {
+        self.calendar_initialized()
+            .ok_or(DstError::ClockNotInitialized)?;
+        if self.reg.cr.read().bkp().bit_is_clear() {
+            return Err(DstError::AlreadyStandardTime);
+        }
+        let time = self.reg.tr.read();
+        if time.ht().bits() == 0 && time.hu().bits() == 0 {
+            return Err(DstError::CannotSubtract);
+        }
+        self.reg
+            .cr
+            .modify(|_, w| w.sub1h().set_bit().bkp().clear_bit());
+        Ok(())
+    }
+
+    /// Configures the wakeup timer to trigger periodically after `interval` seconds
```

Comment:
I think the intention is to trigger periodically *every* `interval` seconds. So I would use the word 'every' since it's repeating